Core/include/Acts/Propagator/ActorConcepts.hpp

```diff
@@ -0,0 +1,105 @@
+    };
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasActWithoutResult = requires(
+    const actor_t& a, propagator_state_t& state, const stepper_t& stepper,
+    const navigator_t& navigator, Args&&... args) {
+  {
+    a.act(state, stepper, navigator, std::forward<Args>(args)...)
+  } -> std::same_as<void>;
+};
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasAct =
+    (!ActorHasResult<actor_t> &&
+     ActorHasActWithoutResult<actor_t, propagator_state_t, stepper_t,
+                              navigator_t, Args...>) ||
+    (ActorHasResult<actor_t> &&
+     ActorHasActWithResult<actor_t, propagator_state_t, stepper_t, navigator_t,
+                           Args...>);
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasAbortWithResult = requires(
+    const actor_t& a, propagator_state_t& state, const stepper_t& stepper,
+    const navigator_t& navigator, typename actor_t::result_type& result,
+    Args&&... args) {
+  {
+    a.checkAbort(state, stepper, navigator, result, std::forward<Args>(args)...)
+  } -> std::same_as<bool>;
+};
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasAbortWithoutResult = requires(
+    const actor_t& a, propagator_state_t& state, const stepper_t& stepper,
+    const navigator_t& navigator, Args&&... args) {
+  {
+    a.checkAbort(state, stepper, navigator, std::forward<Args>(args)...)
+  } -> std::same_as<bool>;
+};
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasAbort =
+    (!ActorHasResult<actor_t> &&
+     ActorHasAbortWithoutResult<actor_t, propagator_state_t, stepper_t,
+                                navigator_t, Args...>) ||
+    (ActorHasResult<actor_t> &&
+     ActorHasAbortWithResult<actor_t, propagator_state_t, stepper_t,
+                             navigator_t, Args...>);
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasActAndAbort =
+    (!ActorHasResult<actor_t> &&
+     ActorHasActWithoutResult<actor_t, propagator_state_t, stepper_t,
+                              navigator_t, Args...> &&
+     ActorHasAbortWithoutResult<actor_t, propagator_state_t, stepper_t,
+                                navigator_t, Args...>) ||
+    (ActorHasResult<actor_t> &&
+     ActorHasActWithResult<actor_t, propagator_state_t, stepper_t, navigator_t,
+                           Args...> &&
+     ActorHasAbortWithResult<actor_t, propagator_state_t, stepper_t,
+                             navigator_t, Args...>);
```

Comment:
This then simplifies to:

```suggestion
template <typename actor_t, typename propagator_state_t, typename stepper_t,
          typename navigator_t, typename... Args>
concept ActorHasActAndAbort = ActorHasAct<T> && ActorHasAbort<T>;
```